src/renderer/components/execute/ParameterContainer.tsx

```diff
@@ -12,28 +16,60 @@ export type Parameter = {
 
 const ParameterContainer = ({
   activeParameters,
-  handleInput,
+  setField,
+  errors,
 }: IParameterContainerProps) => {
+  const [parameterValues, setParameterValues] = useState<Parameter>({});
+
+  const handleChange = (inputValue: string, attribute: string) => {
+    setParameterValues({
+      ...parameterValues,
+      [attribute]: inputValue,
+    });
+  };
+
+  useEffect(() => {
+    const defaultValues: Parameter = {};
+    activeParameters.forEach((item) => {
+      defaultValues[item.name] = '';
+    });
+    setParameterValues(defaultValues);
+  }, [activeParameters]);
```

Comment:
shouldn't this be parameterValues?

Reply:
I'm kind of assuming that's what the second useEffect is for, but isn't it possible to combine the two? I'm not really sure what the one below this does

Reply:
This useeffect updates the parameter values state when the active parameters change. The other one fires when parameter values changes, which happens in two places so it needs its own use effect. You're right that this useeffect will fire that other one as well 100% of the time